text/0000-java-17.md

```diff
@@ -0,0 +1,83 @@
+
+- Start Date: 2023-04-19 
+- RFC PR:
+- FOLIO Issue:
+
+# Java 17
+
+## Summary
+
+From Quesnelia (R2 2023) release on version 17 must be used for JDK and JRE.
+
+## Motivation
+
+### FOLIO Java libraries
+
+FOLIO maintains many Java libraries:
+https://wiki.folio.org/display/REL/pom.xml+dependencies
+
+Some of them are Spring based and have already migrated to Java 17
+because Spring no longer supports Java 11.
+
+As FOLIO still officially supports Java 11 all other FOLIO Java
+libraries must be for Java 11. Java 17 modules can use Java 11
+libraries, but Java 11 modules cannot use Java 17 libraries.
+
+Dropping Java 11 support allows Java libraries to upgrade to Java 17.
+
+### 3rd party Java libraries
+
+Spring no longer supports Java 11. It is likely that other 3rd party
+Java libraries will also drop Java 11 support soon.
+
+### Docker containers
+
+FOLIO DevOps and the FOLIO security team currently need to maintain and
+monitor two JRE containers and two Jenkins JDK containers:
+https://github.com/folio-org/folio-tools/tree/master/folio-java-docker
+https://github.com/folio-org/folio-tools/tree/master/jenkins-slave-docker
+
+Dropping the Java 11 containers reduces the workload.
+
+### Java 11 availability
+
+End of availablility for Java 11 is at least October 2024:
+https://adoptium.net/de/support/
+
+The first flower release that exceeds this date will probably be
+Rocket (R1 2024) because its support period ends around November 2024
+if FOLIO follows a similar cadence to previous releases, see FOLIO support
+policy:
+https://wiki.folio.org/display/PC/PC+Supports+long-term+release+and+regular+release+recommendations
+
+## Detailed Explanation/Design
+
+All Java/JDK/JRE based modules and software libraries are affected.
+
+Releases for Quesnelia and all further flower released must use Java 17
+for build (JDK) and for runtime (JRE).
+
+Orchid and Poppy releases may continue to use Java 11 until these
+flower releases reach their end of life (end of official support period).
+
+## Risks and Drawbacks
+
+The migration of Spring based modules has shown that it's little effort and low risk.
```

Comment:
It was my understanding that the Spring upgrades were delayed by a release cycle, and some issues were introduced during those upgrades, due to the complexity of the upgrade.

Is my recollection of this incorrect?

Reply:
Ok, thank you for clarifying.

I think it could be worth making the distinction more explicit, so that folks who are aware that the combined upgrade took a long time and introduced issues, don't think that came from the Java 17 part.